code/modules/clothing/suits/_suits.dm

```diff
@@ -46,3 +54,53 @@
 		if(DIGITIGRADE in H.dna.species.species_traits)
 			adjusted = DIGITIGRADE_STYLE
 		H.update_inv_w_uniform()
+
+/obj/item/clothing/suit/attackby(obj/item/I, mob/user, params)
+	if(!attach_badge(I, user))
+		return ..()
+
+/obj/item/clothing/suit/proc/attach_badge(obj/item/I, mob/user)
+	. = FALSE
+	if(!istype(I, /obj/item/badge))
+		return
+	if(user && !user.temporarilyRemoveItemFromInventory(I))
+		return
+	var/obj/item/badge/B = I
+	if(!B.try_attach(src, user))
+		return
+	if(user)
+		to_chat(user, span_notice("You attach [I] to [src]."))
+	var/accessory_state = attached_badge.icon_state
+	badge_overlay = mutable_appearance(attached_badge.mob_overlay_icon, "[accessory_state]")
+	badge_overlay.alpha = attached_badge.alpha
+	badge_overlay.color = attached_badge.color
+	if(ishuman(loc))
+		var/mob/living/carbon/human/H = loc
+		H.update_inv_w_uniform()
+		H.update_inv_wear_suit()
+	return TRUE
+
+/obj/item/clothing/suit/AltClick(mob/user)
+	if(..())
+		return 1
+
+	if(!istype(user) || !user.canUseTopic(src, BE_CLOSE, ismonkey(user)))
```

Comment:
Why check for monkey instead of just letting people Alt click the suit?

Reply:
I do not know, it is copy-pasted from `/obj/item/clothing/accessory`

Reply:
it's a dexterity check